stage0/src/kernel.rs

```diff
@@ -0,0 +1,170 @@
+//
+// Copyright 2023 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::fw_cfg::{check_memory, check_non_overlapping, find_suitable_dma_address, FwCfg};
+use core::{ffi::CStr, slice};
+use goblin::{
+    container::{Container, Ctx, Endian},
+    elf::{header, Elf, ProgramHeader},
+    elf64::program_header::PT_LOAD,
+};
+use oak_linux_boot_params::BootE820Entry;
+use x86_64::{PhysAddr, VirtAddr};
+
+/// The default start location and entry point for the kernel if a kernel wasn't supplied via the
+/// QEMU fw_cfg device.
+const DEFAULT_KERNEL_START: u64 = 0x200000;
+
+/// The default size for the kernel if a kernel wasn't supplied via the QEMU fw_cfg device.
+///
+/// This is an arbitrary value, just to ensure it is non-zero.
+const DEFAULT_KERNEL_SIZE: usize = 0x100000;
+
+/// The file path used by Stage0 to read the kernel from the fw_cfg device.
+const KERNEL_FILE_PATH: &[u8] = b"opt/stage0/elf_kernel\0";
```

Comment:
I feel like this should be in `kernel.rs` similar to all the other paths (like for the initramfs or arguments).

Reply:
Oops, that is actually here.